dfpl/options.py

```diff
@@ -1034,141 +1226,151 @@ def parseTrainGnn(parser: argparse.ArgumentParser) -> None:
     )
     files_args.add_argument(
-        "--fnnModelDir",
+        "--calibration_bond_descriptors_path",
         type=str,
-        metavar="DIR",
-        help="The directory where the full model of the fnn is loaded from. "
-        "Provide a full path here.",
-        default=argparse.SUPPRESS,
+        help="Path to the extra bond descriptors that will be used as bond features to featurize a given molecule.",
     )
 
+    general_args.add_argument(
+        "--drop_extra_columns",
+        action="store_true",
+        help="Whether to drop all columns from the test data file besides the SMILES columns and the new prediction columns.",
+    )
 
-def parsePredictGnn(parser: argparse.ArgumentParser) -> None:
-    general_args = parser.add_argument_group("General Configuration")
-    data_args = parser.add_argument_group("Data Configuration")
-    files_args = parser.add_argument_group("Files")
-    training_args = parser.add_argument_group("Training Configuration")
-    files_args.add_argument(
-        "-f",
-        "--configFile",
-        metavar="FILE",
+    uncertainty_args.add_argument(
+        "--uncertainty_method",
         type=str,
-        help="Input JSON file that contains all information for training/predicting.",
-        default=argparse.SUPPRESS,
+        choices=[
+            "mve",
+            "ensemble",
+            "evidential_epistemic",
+            "evidential_aleatoric",
+            "evidential_total",
+            "classification",
+            "dropout",
+            "spectra_roundrobin",
+            "dirichlet",
+        ],
+        help="The method of calculating uncertainty.",
     )
-    general_args.add_argument(
-        "--gpu",
-        type=int,
-        metavar="INT",
-        choices=list(range(torch.cuda.device_count())),
-        help="Which GPU to use",
+    uncertainty_args.add_argument(
+        "--calibration_method",
+        type=str,
+        nargs="+",
+        choices=[
+            "zscaling",
+            "tscaling",
+            "zelikman_interval",
+            "mve_weighting",
+            "platt",
+            "isotonic",
+        ],
+        help="Methods used for calibrating the uncertainty calculated with uncertainty method.",
     )
-    general_args.add_argument(
-        "--no_cuda", action="store_true", default=False, help="Turn off cuda"
+    uncertainty_args.add_argument(
+        "--individual_ensemble_predictions",
+        action="store_true",
+        default=False,
+        help="Whether to save individual ensemble predictions.",
     )
-    general_args.add_argument(
-        "--num_workers",
+    uncertainty_args.add_argument(
+        "--evaluation_methods",
+        type=str,
+        nargs="+",
+        help="The methods used for evaluating the uncertainty performance if the test data provided includes targets. Available methods are [nll, miscalibration_area, ence, spearman] or any available classification or multiclass metric.",
```

Comment:
One more request from my side. In the Galaxy wrappers we can distinguish a `label` and `help` text. My idea would be to use the first sentence of the argument `help` as label and the rest as help. 

```suggestion
        help="Methods used for evaluating the uncertainty performance. Only used if the test data provided includes targets. Available methods are [nll, miscalibration_area, ence, spearman] or any available classification or multiclass metric.",
```

Reply:
Okay, I will do that later today.

Reply:
> So should all have a help text? or if it only 1 sentence(=label) works? 

1 short sentence is fine. Should just be as short as possible. Then the parameter will only have a label in Galaxy.

> And any other things I should avoid or use?(like use choices instead of nargs)

`nargs` and `choices` are two different things (to the bst of my knowledge).  With `nargs` you can allow to specify the same argument multiple times. Then you will get a list of argument values. With `choices` you just limit the possible values.